registry/base_configuration_listener.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package registry
+
+import (
+	perrors "github.com/pkg/errors"
+)
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/config"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/config_center"
+	"github.com/apache/dubbo-go/remoting"
+)
+
+type BaseConfigurationListener struct {
+	configurators           []config_center.Configurator
+	dynamicConfiguration    config_center.DynamicConfiguration
+	defaultConfiguratorFunc func(url *common.URL) config_center.Configurator
+}
+
+func (bcl *BaseConfigurationListener) Configurators() []config_center.Configurator {
+	return bcl.configurators
+}
+func (bcl *BaseConfigurationListener) InitWith(key string, listener config_center.ConfigurationListener, f func(url *common.URL) config_center.Configurator) {
+	bcl.dynamicConfiguration = config.GetEnvInstance().GetDynamicConfiguration()
+	if bcl.dynamicConfiguration == nil {
+		//set configurators to empty
+		bcl.configurators = []config_center.Configurator{}
+		return
+	}
+	bcl.defaultConfiguratorFunc = f
+	bcl.dynamicConfiguration.AddListener(key, listener)
+	if rawConfig, err := bcl.dynamicConfiguration.GetConfig(key, config_center.WithGroup(constant.DUBBO)); err != nil {
+		//set configurators to empty
+		bcl.configurators = []config_center.Configurator{}
+		return
+	} else if len(rawConfig) > 0 {
```

Comment:
I think u can delete the else as follows,

if rawConfig, err := bcl.dynamicConfiguration.GetConfig(key, config_center.WithGroup(constant.DUBBO)); err != nil {
		//set configurators to empty
		bcl.configurators = []config_center.Configurator{}
		return
	}

         if len(rawConfig) > 0 {
           ...
         }

Reply:
If no else here, I can not access the variable rawConfig produced above.

Reply:
Pls see it carefully. What I mean is that just 'deleting the else word'.

Reply:
![image](https://user-images.githubusercontent.com/13016899/64581807-1e78f000-d3be-11e9-98c0-016258453c2a.png)

Reply:
@AlexStocks I means this if I delete else ,then I can not access the variable rawConfig.